### PR TITLE
Merged ``Tessellation`` and ``Divide`` into ``Sampling`` class.

### LINKED ISSUE #312: Refactor ``Tessellation`` and ``Divide``
Merged ``Tessellation`` and ``Divide`` into ``Sampling.Curve`` and ``Sampling.Surface``.

### PR DESCRIPTION
**Pull request by sonomirco**
### What type of PR is this? (check all applicable)

- [X] ♻️ Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

### Description

This PR starts the transition to the new structure and namespaces.

### Related Tickets & Documents

This PR closes #312 

### Added tests?

- [ ] 👍 yes
- [X] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📓 docs
- [X] 🙅 no documentation needed

### CODE REVIEW

File: src/GShark.Test.XUnit/Sampling/CurveTests.cs [hunk 1]
```diff
@@ -298,5 +296,95 @@ public void It_Calculates_Rotation_Minimized_Frames_Along_Curve()
...
+            Point3 p1 = new Point3(0, 0, 0);
+            Point3 p2 = new Point3(10, 0, 0);
+            Line ln = new Line(p1, p2);
+
+            // Act
+            var (tValues, pts) = ln.ToNurbs().AdaptiveDivision();
+
+            // Arrange
+            pts.Count.Should().Be(tValues.Count).And.Be(2);
+            pts[0].DistanceTo(p1).Should().BeLessThan(GSharkMath.MaxTolerance);
```
Comment (cesarecaoduro) [inline src/GShark.Test.XUnit/Sampling/CurveTests.cs:354]: Does it fail with the min tolerance?

File: src/GShark.Test.XUnit/Sampling/CurveTests.cs [hunk 2]
```diff
@@ -298,5 +296,95 @@ public void It_Calculates_Rotation_Minimized_Frames_Along_Curve()
...
+            List<Point3> pts = new List<Point3> { p1, p2, p3, p4, p5 };
+            Polyline poly = new Polyline(pts);
+
+            // Act
+            (List<double> tValues, List<Point3> pts) result = poly.ToNurbs().AdaptiveDivision();
+
+            // Arrange
+            result.pts.Count.Should().Be(result.tValues.Count).And.Be(5);
+            result.pts[0].DistanceTo(p1).Should().BeLessThan(GSharkMath.MaxTolerance);
+            result.pts[result.pts.Count - 1].DistanceTo(p5).Should().BeLessThan(GSharkMath.MaxTolerance);
```
Comment (cesarecaoduro) [inline src/GShark.Test.XUnit/Sampling/CurveTests.cs:376]: Does it fail with min or mid tolerance?

File: src/GShark.Test.XUnit/Sampling/SurfaceTest.cs [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+            Point3 expectedPtLeft = new Point3(5.0, 3.333333, 1.444444);
+            Point3 expectedPtRight = new Point3(5.0, 8.181818, 3.545455);
+
+            // Act
+            NurbsSurface[] surfaces = surface.SplitAt(0.5, SplitDirection.V);
+            Point3 evaluatePtLeft = surfaces[0].PointAt(0.5, 0.5);
+            Point3 evaluatePtRight = surfaces[1].PointAt(0.5, 0.5);
+
+            // Assert
+            evaluatePtLeft.DistanceTo(expectedPtLeft).Should().BeLessThan(GSharkMath.MaxTolerance);
```
Comment (cesarecaoduro) [inline src/GShark.Test.XUnit/Sampling/SurfaceTest.cs:57]: Does it fail with min or mid tolerance?

File: src/GShark.Test.XUnit/Sampling/SurfaceTest.cs [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+            evaluatePtRight.DistanceTo(expectedPtRight).Should().BeLessThan(GSharkMath.MaxTolerance);
+
+            surfaces[0].Weights.Should().BeEquivalentTo(weightsLeft);
+            surfaces[1].Weights.Should().BeEquivalentTo(weightsRight);
+
+            _ = surfaces[0].LocationPoints.Select((pts, i) => pts.Select((pt, j) =>
+                pt.EpsilonEquals(surfacePtsLeft[i][j], GSharkMath.MaxTolerance).Should().BeTrue()));
```
Comment (cesarecaoduro) [inline src/GShark.Test.XUnit/Sampling/SurfaceTest.cs:64]: Does it fail with min or mid tolerance?

File: src/GShark.Test.XUnit/Sampling/SurfaceTest.cs [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+            Point3 expectedPtTop = new Point3(2.894737, 5.789474, 2.578947);
+            Point3 expectedPtBottom = new Point3(7.105263, 5.789474, 2.157895);
+
+            // Act
+            NurbsSurface[] surfaces = surface.SplitAt(0.5, SplitDirection.U);
+            Point3 evaluatePtTop = surfaces[0].PointAt(0.5, 0.5);
+            Point3 evaluatePtBottom = surfaces[1].PointAt(0.5, 0.5);
+
+            // Assert
+            evaluatePtTop.DistanceTo(expectedPtTop).Should().BeLessThan(GSharkMath.MaxTolerance);
```
Comment (cesarecaoduro) [inline src/GShark.Test.XUnit/Sampling/SurfaceTest.cs:111]: Does it fail with min or mid tolerance?

File: src/GShark.Test.XUnit/Sampling/SurfaceTest.cs [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+                new Point3(6.95122, 8.04878, 3)
+            };
+
+            // Act
+            NurbsSurface[] splitSrf = surface.SplitAt(0.5, SplitDirection.Both);
+            var ptsAt = splitSrf.Select(s => s.PointAt(0.5, 0.5));
+
+            // Assert
+            splitSrf.Length.Should().Be(4);
+            _ = ptsAt.Select((pt, i) => pt.DistanceTo(expectedPts[i]).Should().BeLessThan(GSharkMath.MaxTolerance));
```
Comment (cesarecaoduro) [inline src/GShark.Test.XUnit/Sampling/SurfaceTest.cs:142]: Does it fail with min or mid tolerance?

File: src/GShark/Geometry/NurbsCurve.cs [hunk 7]
```diff
@@ -463,11 +463,36 @@ public NurbsCurve Offset(double distance, Plane pln)
...
+        /// <summary>
+        /// Samples a curve in an adaptive way. <br/>
+        /// <em>Corresponds to this algorithm http://ariel.chronotext.org/dd/defigueiredo93adaptive.pdf <br/>
+        /// https://www.modelical.com/en/grasshopper-scripting-107/ </em>
+        /// </summary>
+        /// <param name="start">The start parameter for sampling.</param>
+        /// <param name="end">The end parameter for sampling.</param>
+        /// <param name="tolerance">Tolerance for the adaptive scheme. The default tolerance is set as (1e-6).</param>
+        /// <returns>A tuple with the set of points and the t parameter where the point was evaluated.</returns>
+        public (List<double> Parameters, List<Point3> Points) AdaptiveDivision(double start, double end, double tolerance = 1e-6)
```
Comment (cesarecaoduro) [inline src/GShark/Geometry/NurbsCurve.cs:491]: Change the value 1e-6 with min tolerance?

File: src/GShark/Sampling/Curve.cs [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+            double sum2 = 0.0;
+            double segmentLength = length;
+
+            while (i < curves.Count)
+            {
+                sum += curveLengths[i];
+
+                while (segmentLength < sum + GSharkMath.Epsilon)
+                {
+                    double t = Analyze.BezierCurveParamAtLength(curves[i], segmentLength - sum2, GSharkMath.MaxTolerance);
```
Comment (cesarecaoduro) [inline src/GShark/Sampling/Curve.cs:67]: Does it work with a smaller tolerance?

File: src/GShark/Geometry/NurbsCurve.cs (hunk 7 above)
Comment (sonomirco) [reply src/GShark/Geometry/NurbsCurve.cs:491]: I can use a value from a static class as a default value.

**Pull request merged**